x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/api_key/10_grant.yml

```diff
@@ -0,0 +1,216 @@
+              "role_descriptors": {
+                "role": {
+                  "index": [
+                    {
+                      "names": ["index"],
+                      "privileges": ["read"]
+                    }
+                  ]
+                }
+              }
+            },
+            "grant_type": "password",
+            "username": "api_key_grant_target_user",
+            "password": "x-pack-test-password-2"
+          }
+  - match: { name: "my-api-key" }
+  - is_true: id
+  - is_true: api_key
+  - is_true: expiration
+  - set: { id: api_key_id }
+  - transform_and_set: { login_creds: "#base64EncodeCredentials(id,api_key)" }
+  - match: { encoded: $login_creds }
+
+  - do:
+      headers:
+        Authorization: ApiKey ${login_creds}
+      security.authenticate: {}
+
+  - match: { username: "api_key_grant_target_user" }
+  - length: { roles: 0 }
+  - match: { authentication_realm.name: "_es_api_key" }
+  - match: { authentication_realm.type: "_es_api_key" }
+  - match: { api_key.id: "${api_key_id}" }
+  - match: { api_key.name: "my-api-key" }
+
+  - do:
+      security.clear_api_key_cache:
+        ids: "${api_key_id}"
+
+  - match: { _nodes.failed: 0 }
+
+---
+"Test grant api key with token":
+  - do:
+      security.get_token:
+        body:
+          grant_type: "password"
+          username: "api_key_grant_target_user"
+          password: "x-pack-test-password-2"
+
+  - match: { type: "Bearer" }
+  - is_true: access_token
+  - set: { access_token: token }
+
+  - do:
+      headers:
+        Authorization: "Basic YXBpX2tleV9ncmFudGVyOngtcGFjay10ZXN0LXBhc3N3b3Jk" # api_key_granter
+      security.grant_api_key:
+        body:  >
+          {
+            "api_key": {
+              "name": "my-api-key",
+              "expiration": "1d",
+              "role_descriptors": {
+                "role": {
+                  "index": [
+                    {
+                      "names": ["index"],
+                      "privileges": ["read"]
+                    }
+                  ]
+                }
+              }
+            },
+            "grant_type": "access_token",
+            "access_token": "$token"
+          }
+  - match: { name: "my-api-key" }
+  - is_true: id
+  - is_true: api_key
+  - is_true: expiration
+  - set: { id: api_key_id }
+  - transform_and_set: { login_creds: "#base64EncodeCredentials(id,api_key)" }
+  - match: { encoded: $login_creds }
+
+  - do:
+      headers:
+        Authorization: ApiKey ${login_creds}
+      security.authenticate: {}
+
+  - match: { username: "api_key_grant_target_user" }
+  - length: { roles: 0 }
+  - match: { authentication_realm.name: "_es_api_key" }
+  - match: { authentication_realm.type: "_es_api_key" }
+  - match: { api_key.id: "${api_key_id}" }
+  - match: { api_key.name: "my-api-key" }
+
+  - do:
+      security.clear_api_key_cache:
+        ids: "${api_key_id}"
+
+  - match: { _nodes.failed: 0 }
+
+---
+"Test grant api key forbidden":
```

Comment:
For this test, I'd suggest we have a 3rd user which has `manage_own_api_key` privilege and assert it cannot grant API keys (not even for itself)

Reply:
Yup, I can add that. I wanted to keep these relatively high level since we cover other corner cases in the Java REST tests already, but happy to make the addition 👍